pkg/federation/get.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2023 The Bestchains Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package federation
+
+import (
+	"context"
+	"fmt"
+	"github.com/bestchains/bc-cli/pkg/common"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubectl/pkg/cmd/get"
+)
+
+func NewFedGetCmd(option common.Options) *cobra.Command {
+
+	defaultPrintFlag := get.NewGetPrintFlags()
+	cmd := &cobra.Command{
+		Use: "fed [FED-NAME]... [-o json/yaml] [--with-org ORG-NAME]",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// get config from getter
+			cfg, err := clientcmd.BuildConfigFromKubeconfigGetter("", common.InKubeGetter)
+			if err != nil {
+				fmt.Fprintln(option.ErrOut, err)
+				return err
+			}
+			cli, err := dynamic.NewForConfig(cfg)
+			if err != nil {
+				fmt.Fprintln(option.ErrOut, err)
+				return err
+			}
+			list := corev1.List{
+				TypeMeta: v1.TypeMeta{
+					Kind:       "List",
+					APIVersion: "v1",
+				},
+				ListMeta: v1.ListMeta{},
+			}
+
+			var obj runtime.Object
+
+			orgName, _ := cmd.Flags().GetString("with-org")
+
+			if orgName == "" {
+				if len(args) == 0 {
+					// get all feds of all orgs
+					orgs, err := cli.Resource(schema.GroupVersionResource{
+						Group:    common.IBPGroup,
+						Version:  common.IBPVersion,
+						Resource: common.OrganizationResource,
+					}).List(context.TODO(), v1.ListOptions{LabelSelector: "user=account"}) // FIXME: Selector
```

Comment:
Do we need to add `LabelSelector`? According to my tests, it is possible to get all the organization information, even if the current user is not the creator of the organization.
@bjwswang

Reply:
@0xff-dev @Lanture1064 well... yes.. we need to list the organizations in which this user is the client/admin

Reply:
@0xff-dev please help with @Lanture1064

Reply:
@Lanture1064  Merge  the code of`connProfile`, which will have `username`  in viper's configuration, the organization information is obtained from the `annotations["bestchains"]` field of the user. Then get the information about the federation to which each organization belongs by get org.

```json
{
  "list": {
    "org1": {},
    "org2": {},
  }
}
```